infoschema/builder.go

```diff
@@ -367,7 +367,10 @@ func (b *Builder) applyTableUpdate(m *meta.Meta, diff *model.SchemaDiff) ([]int6
 	dbInfo := b.getSchemaAndCopyIfNecessary(roDBInfo.Name.L)
 	var oldTableID, newTableID int64
 	switch diff.Type {
-	case model.ActionCreateTable, model.ActionCreateSequence, model.ActionRecoverTable:
+	case model.ActionCreateSequence, model.ActionRecoverTable:
+		newTableID = diff.TableID
+	case model.ActionCreateTable:
+		oldTableID = diff.OldTableID
```

Comment:
How about upgrading TiDB？
TiDB 1（foreign key version is 1, owner）：this TiDB will update diffInfo.
TiDB 2（foreign key version is 2, not owner）： this TiDB can't do this logic in line373

Is it OK？

Reply:
Nice catch!

Reply:
Since the cluster-index feature also has similar problem, we chose to prevent DDL execution during the upgrade process to avoid this issue.